js-core-2/lesson5.md

```diff
@@ -33,21 +32,107 @@ Find all the bad design issues with this function
+
+Now we will step into a new level of abstractions - Objects.
+
+> The basic idea of OOP is that we use objects to model real world things that we want to represent inside our programs, and/or provide a simple way to access functionality that would otherwise be hard or impossible to make use of.
+
+## OO overview
+
+### Defining an object template
+
+![](https://mdn.mozillademos.org/files/13889/person-diagram.png)
+
+### Creating actual objects
+
+![](https://mdn.mozillademos.org/files/15163/MDN-Graphics-instantiation-2-fixed.png)
+
+### Specialist classes
+
+![](https://mdn.mozillademos.org/files/13881/MDN-Graphics-inherited-3.png)
+
+
+> Some people argue that JavaScript is not a true object-oriented language — for example, it doesn't have a class statement for creating classes, unlike many OO languages. JavaScript, instead, uses special functions called constructor functions to define objects and their features. They are useful because you'll often come across situations in which you don't know how many objects you will be creating; constructors provide the means to create as many objects as you need in an effective way, attaching data and functions to them as required.
+
+> Exercise: On a paper, **model** how you would create a system to represent an **animal**  - what class do you have, what objects, what properties, create specialised classes for
+(ToDo: better exercise)
+
+## Object literals
+There are several ways to create objects - we've already used one before. It is called `object literals`
+
+`var person = {}`
+
+The line before created an empty object. Let's create a more sophisticated object
+
+```javascript
+var person = {
+  name: ['Bob', 'Smith'],
+  age: 32,
+  gender: 'male',
+  interests: ['music', 'skiing'],
+  bio: function() {
+    alert(this.name[0] + ' ' + this.name[1] + ' is ' + this.age + ' years old. He likes ' + this.interests[0] + ' and ' + this.interests[1] + '.');
+  },
+  greeting: function() {
+    alert('Hi! I\'m ' + this.name[0] + '.');
+  }
+};
+```
+
+So what is going on here? Well, an object is made up of multiple members, each of which have a name (e.g. name and age above), and a value (e.g. ['Bob', 'Smith'] and 32). Each name/value pair must be separated by a comma, and the name and value in each case are separated by a colon. The syntax always follows this pattern:
+
+The value of an object member can be pretty much anything — in our person object we've got a string, a number, two arrays, and two functions. The first four items are `data items`, and are referred to as the object's properties. The last two items are functions that allow the object to do something with that data, and are referred to as the object's `methods`.
+
+> Methods are functions that allow the object to do something with its data.
+
+> The *this* keyword refers to the current object the code is being written inside — so in this case this is equivalent to person. So why not just write person instead? 
```

Comment:
yup, I copy/pasted that explanation but it's not accurate at all. We might come back to *this* again, but for now, it's enough to say it refers to the context, which is person in this case.